### PR TITLE
Implement rebootstrap-raft command

### PR DESCRIPTION
**Pull request by babbageclunk**
This will generate an empty raft directory with servers matching the MongoDB replicaset.

### CODE REVIEW

File: src/rebootstrap-raft/main.go [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+}
+
+func (c *rebootstrapCommand) dial() (*mgo.Session, error) {
+	info := &mgo.DialInfo{
+		Addrs: []string{net.JoinHostPort(c.hostname, c.mongoPort)},
+	}
+	if c.username != "" {
+		info.Database = "admin"
+		info.Username = c.username
+		info.Password = c.password
```
Comment (wallyworld) [inline src/rebootstrap-raft/main.go]: should we warn or error if password is not supplied?

**Review by wallyworld — APPROVED** [review]
Looks ok, assuming it tests live.
Could we add a --dry-run mode that allow it to be run on any server even if the raft dir is still there. It would allow to user to run it anywhere to see what would be done on the problematic server.

File: src/rebootstrap-raft/main.go (hunk 1 above)
Comment (babbageclunk) [reply src/rebootstrap-raft/main.go]: Yeah, I think this and machineID should be required parameters.

**Comment by babbageclunk** [comment]
Good idea, I'll add dry-run - it'll just get the replicaset members and show the raft servers it would add.

**Comment by jameinel** [comment]
Thanks for doing this. As a side comment, I wonder if it would be easier if
we integrated this as "jujud rebootstrap-raft".
That way the functionality is part of the core tool. And we don't have to
worry about bit rot or finding the tool.

Just a thought. I know mgopurge suffers a bit from not being very
discoverable or integrated.

John
=:->

**Comment by mitechie** [comment]
Just a +1 to having it available. I like how we have dump-db behind a feature flag but then available for use. I think following that model would be ideal and set a standard pattern for "hopefully never needed operator tools" going forward.

**Comment by babbageclunk** [comment]
Yeah, having it as `jujud rebootstrap-raft` does make sense - it would also mean a lot of the code in here that's ripped from the juju codebase could just be used directly, and we could parse the agent conf rather than having the operator get the password. 
I'll do that once I've got a moment - hopefully after sorting out the other raft issue...

**Pull request merged**